xivo_bus/mixins.py

```diff
@@ -233,49 +250,82 @@ def __on_message_received(self, body, message):
         finally:
             message.ack()
 
+    def on_connection_error(self, exc, interval):
+        self.log.error(
+            'Broker connection error: %s, trying to reconnect in %s seconds...',
+            exc,
+            interval,
+        )
+        if self.should_stop:
+            raise SystemExit
```

Comment:
I can understand this usage on Threadable context, but this mixin can be use without thread. Is it safe/good practice do raise SystemExit in *normal* context?

Reply:
Always executed in a Thread